packages/web/src/components/edit/fields/CollectionTrackField.tsx

```diff
@@ -16,15 +16,19 @@ type CollectionTrackFieldProps = {
   disableDelete: boolean
 }
 
+type CollectionTrackWithOverride = (TrackForEdit | TrackForUpload) & {
+  override: boolean
+}
+
 export const CollectionTrackField = (props: CollectionTrackFieldProps) => {
   const { disableDelete = false, index, remove } = props
   const { playingPreviewIndex, stopPreview } = useContext(UploadPreviewContext)
-  const [{ value: track }] = useField<CollectionTrackForUpload>(
+  const [{ value: track }] = useField<CollectionTrackWithOverride>(
     `tracks.${index}`
   )
 
   const [{ value: metadata }, , { setValue }] = useField<
-    CollectionTrackForUpload['metadata']
+    CollectionTrackWithOverride['metadata']
```

Comment:
i don't quite understand what's going on here

Reply:
`CollectionTrackForUpload` was only extending `TrackForUpload` and adding the override flag. But what actually comes into this component is a `(TrackForUpload | TrackForEdit)`. And since we only use `override` in this component, I just made a type above to represent it.